### ISSUE TITLE
[JENKINS-66490] ERROR trying to upgrade jenkins to 2.303.1 yields message: daemonize is needed

### ISSUE THREAD

**Issue by timja-bot** [timja-org/jenkins-gh-issues-poc-11-04#2009]
<p>hi,</p>

<p> </p>

<p>i'm trying to upgrade jenkins using the redhat package from:</p>

<p><a href="http://mirrors.jenkins-ci.org/redhat-stable/jenkins-2.303.1-1.1.noarch.rpm" class="external-link" target="_blank" rel="nofollow noopener">http://mirrors.jenkins-ci.org/redhat-stable/jenkins-2.303.1-1.1.noarch.rpm</a></p>

<p> </p>

<p>the following set of commands yields an installation error:</p>

<p>================================</p>

<p>-bash-4.2$ sudo rpm -Uvh jenkins-2.303.1-1.1.noarch.rpm<br/>
warning: jenkins-2.303.1-1.1.noarch.rpm: Header V4 RSA/SHA512 Signature, key ID 45f2c3d5: NOKEY<br/>
error: Failed dependencies:<br/>
 daemonize is needed by jenkins-2.303.1-1.1.noarch</p>

<p>================================</p>

<p> </p>

<p>for the past couple of years (at least 5 years) I was able to do so.</p>

<p>did something change in this rpm?</p>

<p>P.S. I installed openjdk11 a week ago - the jenkins controller is working properly.</p>

---
<details><summary><i>Originally reported by <a title="Amit Dar" href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=amidar">amidar</a>, imported from: <a href="https://issues.jenkins.io/browse/JENKINS-66490" target="_blank">ERROR trying to upgrade jenkins to 2.303.1 yields message: daemonize is needed</a></i></summary>
<i><ul>
<li><b>assignee</b>: <a title="Mark Waite" href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=markewaite">markewaite</a>
<li><b>status</b>: Closed
<li><b>priority</b>: Critical
<li><b>resolution</b>: Not A Defect
<li><b>resolved</b>: 2021-08-31T21:01:19+00:00
<li><b>imported</b>: 2025-11-04
</ul></i>
</details>

**Comment by timja-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=markewaite">markewaite</a>:</i>
<p>Refer to the installation instructions at <a href="https://www.jenkins.io/doc/book/installing/linux/#red-hat-centos" class="external-link" target="_blank" rel="nofollow noopener">https://www.jenkins.io/doc/book/installing/linux/#red-hat-centos</a> for the current install instructions.  The Jenkins rpm package now requires the <tt>daemonize</tt> package that is available from the epel-release repository.</p>

<p>Thanks for detecting that we didn't include that change in the Jenkins 2.303.1 upgrade guide.  It needs to be included there, in addition to the inclusion in the installation instructions.</p>

**Comment by timja-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=amidar">amidar</a>:</i>
<p>thanks <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=markewaite" class="user-hover" rel="markewaite">markewaite</a>!</p>

<p>since I'm installing it on an air-gapped centos 6 server, can you please make sure the instructions contain exactly what is needed - package wise?</p>

<p>also - is there any additional configuration which needs to be done prior to the upgrade itself.</p>

<p>thanks.</p>

**Comment by timja-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=markewaite">markewaite</a>:</i>
<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=amidar" class="user-hover" rel="amidar">amidar</a> I'm sorry, but I don't test CentOS 6 and don't intend to document or test for CentOS 6.  CentOS 6 <a href="https://wiki.centos.org/About/Product" class="external-link" target="_blank" rel="nofollow noopener">end of life</a> was November 2020.</p>

<p>The command line git version included in CentOS 6 has several use cases that do not work with the Jenkins git plugin.  CentOS 6 does not receive maintenance updates.</p>

**Comment by timja-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=amidar">amidar</a>:</i>
<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=markewaite" class="user-hover" rel="markewaite">markewaite</a> fair enough.</p>

<p> </p>

<p>how about CentOS 7? </p>

<p>I guess the time for server upgrade has come... still I need to know how to treat this air gapped environment... (up until now the jenkins rpm was all I needed...)</p>

**Comment by timja-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=markewaite">markewaite</a>:</i>
<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=amidar" class="user-hover" rel="amidar">amidar</a> The existing instructions work for CentOS 7 with an internet connection.  You'll need to explore for yourself to see what is required in your air-gapped environment.</p>

**Comment by timja-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=markewaite">markewaite</a>:</i>
<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=phreakadelle" class="user-hover" rel="phreakadelle">phreakadelle</a> the dependency in the rpm file can only be satisfied with an rpm package for daemonize.  Installing from source won't do it because the RPM package manager won't realize that the dependency is satisfied.  You might try enabling the epel-release repository (if they still support CentOS 6) and then installing.</p>

**Comment by timja-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=amidar">amidar</a>:</i>
<p>I eventually updated the air gapped Jenkins server on our site. it has centos 6.4 and I used the following packages:</p>

<p><a href="https://fedora-archive.ip-connect.info/epel/6/x86_64/epel-release-6-8.noarch.rpm" class="external-link" target="_blank" rel="nofollow noopener">https://fedora-archive.ip-connect.info/epel/6/x86_64/epel-release-6-8.noarch.rpm</a></p>

<p><a href="https://fedora-archive.ip-connect.info/epel/6/x86_64/Packages/d/daemonize-1.7.3-1.el6.x86_64.rpm" class="external-link" target="_blank" rel="nofollow noopener">https://fedora-archive.ip-connect.info/epel/6/x86_64/Packages/d/daemonize-1.7.3-1.el6.x86_64.rpm</a></p>

<p> </p>
